src/authentication-strategies/JWT.strategy.ts

```diff
@@ -0,0 +1,30 @@
+const jwt = require('jsonwebtoken');
+import {promisify} from 'util';
+const verifyAsync = promisify(jwt.verify);
+// Consider turn it to a binding
+const SECRET = 'secretforjwt';
+import {Request, HttpErrors} from '@loopback/rest';
+import {UserProfile} from '@loopback/authentication';
+import * as _ from 'lodash';
+import {AuthenticationStrategy} from './authentication.strategy';
+
+export class JWTStrategy implements AuthenticationStrategy {
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    // there is a discussion regarding how to retrieve the token,
+    // see comment https://github.com/strongloop/loopback-next/issues/1997#issuecomment-451054806
+    const token = request.query.token || request.headers['authorization'];
+    if (token) {
+      try {
+        const decoded = await verifyAsync(token, SECRET);
+        let user = _.pick(decoded, ['id', 'email', 'firstName']);
+        (user as UserProfile).name = user.firstName;
+        delete user.firstName;
+        return user;
+      } catch (err) {
+        throw new HttpErrors.Unauthorized('Could not decode the JWT token!');
```

Comment:
This is discarding information about the original error: the reason why the token could not be decoded, the stack trace pointing to the problematic place.

I am proposing to use the following solution:

```ts
} catch (err) {
  Object.assign(err, {
    code: 'INVALID_ACCESS_TOKEN',
    statusCode: 401,
  });
  throw err;
}
```